### ISSUE TITLE
Need to have a mono spaced Thai font

### ISSUE THREAD

**Issue by budrwp** [IBM/plex#308]
I have sent the Plex fonts to many customers in Thailand, and they love the Thai Plex font, BUT, most cannot use the font, because they need a mono space Thai font for business applications.  Are there any plans to create a mono spaced Thai Plex font?  Thanks, and I'd like to help in any way if possible.
Bud

**Comment by mjabbink** [comment]
I will look into a Thai mono. Have not seen one as of yet. What is the use case for these business applications?

**Comment by budrwp** [comment]
Hello Mike,Thanks for writing back!
YES, these are business use cases. I am Worldwide Technical for my product, Content Manager OnDemand. (CMOD)  And many of my customers in Thailand are always having problems viewing their invoices, statements, etc. Because they don't have a "good" mono spaced font. They really need a mono spaced font for applications where they are filling in forms with lots of numbers. So they end up using many different cheap fonts, so it is different from on customer to customer. They are not happy because the data looks bad or the data does not fit well in they forms.
My product archives their data, then later when they retrieve the data, we don't have the font they where using, and the problems start....

I try and explain to them, it is not our product (CMOD), it is their font, but as you know, not too many people understand fonts. I have worked with fonts for 30+ years because I deal with AFP (Advanced Function Presentation) datastream which uses AFP fonts....which is another issue and story.  That would be nice if someone converted all the Plex fonts into AFP fonts! Maybe I'll look into that. AFP started many years ago, before outline fonts and they used raster fonts, and raster fonts don't work for viewing, so when someone tries to view AFP they have to "map" the AFP fonts to a font on their PC (outline fonts) and they never match one for one.

AFP can now use outline fonts, but the font has to be wrapped in AFP structured fields (code).  There used to be a tool to convert a TT or OTF font to AFP, I'll see if I can find the utility.
Sorry, got off subject.  Please... if you can help with a mono spaced Thai font. That would be wonderful.  I'd told some customers in Thailand about Plex and they have used Plex, and they think it is very nice, BUT... they need mono space.
PS. My wife is Thai....I spent so much time in Thailand....

    On Thursday, June 11, 2020, 12:10:30 AM EDT, Mike Abbink <notifications@github.com> wrote:  
 
 


I will look into a Thai mono. Have not seen one as of yet. What is the use case for these business applications?

**Comment by JapanYoshi** [comment]
For reference, I found pictures of text from a Thai typewriter. (Yes, they exist!)
![fig.1](http://3.bp.blogspot.com/-lN6L5xOWyx8/TyN71G98qXI/AAAAAAAAK3s/vEy31H8B8YM/s1600/Thai+writing.jpg)
![fig.2](https://cadsondemak.com/wp-content/uploads/2020/02/monospace_article_a2.jpg)

Not entirely sure what this says, but it could be a clue for an ideal Thai font
![fig.3](https://cadsondemak.com/wp-content/uploads/2020/02/monospace_article_a3.jpg)

**Comment by mjabbink** [comment]
@JapanYoshi we have Plex Thai already but are looking into a monospace version.

**Comment by JapanYoshi** [comment]
Yes, I know. That's why I got pictures from a typewriter; Thai typewriters are fixed pitch like English ones..

**Comment by mjabbink** [comment]
I was referencing your statement “ Not entirely sure what this says, but it could be a clue for an ideal Thai font” and it was accompanied by an image that is not a Mono. But thank you for sending typewriter image

**Comment by cwt** [comment]
There are several Thai monospace  fonts in this project https://github.com/tlwg/fonts-tlwg such as `TlwgMono` `TlwgTypo` `TlwgTypist` `TlwgTypewriter`. And also `ArundinaSansMono` in https://github.com/tlwg/fonts-arundina

However, the look and feel of them are not quite compatible with IBM Plex Thai, but I think you may get some hints of how Thai monospace should be.

**Comment by Mark2Mark** [comment]
As one of the 2 designers of IBM Plex Thai (and also designer of 2 monospaced coding fonts) I am always up for adding the monospaced Thai whenever it’s issued. Years ago I collected a bunch of receipts in Thailand which monospaced fonts on them, is I always longed for adding Thai to my existing monospaced fonts. 

Just my 2 cents to the discussion.

**Comment by mjabbink** [comment]
All. I do not disagree with a Thai Mono not being a bass ass addition. However, I’m plagued with tight budgets and priorities. Currently we are wrapping up Chinese TC & SC, expanding Japanese, Math, and I’m eating for budget approval for Kannada, if I even get that approved. This said, I’m adding to the wish list. Would we do looped or non-looped? My guess is looped but let me know.

**Comment by cwt** [comment]
@mjabbink, I my opinion (and my experience as a Thai coder / Linux admin), 'looped' is considered more formal, while 'non-looped' may be used for headings or large signs, but it's not really appropriate for a lot of characters for reading, for example, coding. So I +1 for 'looped'.

**Comment by mjabbink** [comment]
That is what I suspected. Thanks for validating @cwt. We have both version in Plex for the very reasons you described.

**Comment by Mark2Mark** [comment]
Ben also told me from his recent trip around Thailand, that he spotted Plex Thai loopless a lot more than the looped one, for the mentioned reasons, Thailand heading towards some modernisations.

From the design perspective, the loopless style could help to get the wide glyphs squeezed into the the pitch, however it _could_ compromise legibility (all monospaced Thai fonts I saw so far are looped), not in general for loopless, but for mono loopless.
On the other hand a looped monospace might especially need lots of hacks and tricks to get loops and stems of wide glyphs into the pitch in the bold weights. I assume it could be necessary to ditch one or 2 of the boldest weights when making a looped monospace Thai.
Both styles have disadvantages and advantages.

It could be a nice sub-project for @ohbendy and me to stick heads together again. Would love to do it.

BTW I am also very happy to see that the Thai fonts are widely used and liked.

**Comment by ohbendy** [comment]
> Ben also told me from his recent trip around Thailand, that he spotted Plex Thai loopless a lot more than the looped one, for the mentioned reasons, Thailand heading towards some modernisations.

Yes, Plex Thai loopless is getting plenty of use, I was generally seeing it a couple of times a day when in Thailand for branding, signage and posters. The looped was more designed for continuous text, so I wouldn't really expect it to feature prominently in the environment, perhaps it's more used online.

I've also seen a few contemporary monospaced Thai designs, which like Mark says, tend to be looped. But let's not forget there's a spectrum between fully looped and fully loopless — like the fantastic, recent Keychron font from Cadson Demak — which gives the advantages of clear letterforms that are more readily monospaceable, that are less confusable with Latin letters, and that work better at small sizes than fully looped letters.

**Comment by cwt** [comment]
@ohbendy 
> The looped was more designed for continuous text

Exactly why I +1 for looped monospaced as I would like to use it for Linux terminal and vim / IDE coding.

**Comment by cwt** [comment]
@Palke Did we already have mono spaced Thai?

**Comment by BoldMonday** [comment]
@cwt I’m afraid not. Thai Monospaced is not planned actually.

**Comment by budrwp** [comment]
I am sorry to hear that no Thai Mono spaced fonts are  planned.  Again, the Thai people would greatly benefit from having a good quality Thai mono font. Please try to find a way to make this happen. Thank You for your help
